labelbox/data/annotation_types/data/tiled_image.py

```diff
@@ -0,0 +1,364 @@
+
+
+class TiledBounds(BaseModel):
+    """ Bounds for a tiled image asset related to the relevant epsg. 
+
+    Bounds should be Point objects. Currently, we support bounds in EPSG 4326.
+
+    If version of asset is 2, these should be [[lat,lng],[lat,lng]]
+    If version of asset is 1, these should be [[lng,lat]],[lng,lat]]
+
+    >>> bounds = TiledBounds(
+        epsg=EPSG.4326,
+        bounds=[Point(x=0, y=0),Point(x=100, y=100)]
+        )
+    """
+    epsg: EPSG
+    bounds: List[Point]
+
+    @validator('bounds')
+    def validate_bounds_not_equal(cls, bounds):
+        first_bound = bounds[0]
+        second_bound = bounds[1]
+
+        if first_bound == second_bound:
+            raise AssertionError(f"Bounds cannot be equal, contains {bounds}")
+        return bounds
+
+    #bounds are assumed to be in EPSG 4326 as that is what leaflet assumes
+    @root_validator
+    def validate_bounds_lat_lng(cls, values):
+        epsg = values.get('epsg')
+        bounds = values.get('bounds')
+        #TODO: look into messaging that we only support 4326 right now. raise exception, not implemented
+
+        if epsg != EPSG.SIMPLEPIXEL:
+            for bound in bounds:
+                lat, lng = bound.y, bound.x
+                if int(lng) not in VALID_LNG_RANGE or int(
+                        lat) not in VALID_LAT_RANGE:
+                    raise ValueError(f"Invalid lat/lng bounds. Found {bounds}. "
+                                     "lat must be in {VALID_LAT_RANGE}. "
+                                     "lng must be in {VALID_LNG_RANGE}.")
+        return values
+
+
+class TileLayer(BaseModel):
+    """ Url that contains the tile layer. Must be in the format:
+
+    https://c.tile.openstreetmap.org/{z}/{x}/{y}.png
+
+    >>> layer = TileLayer(
+        url="https://c.tile.openstreetmap.org/{z}/{x}/{y}.png",
+        name="slippy map tile"
+        )
+    """
+    url: str
+    name: Optional[str] = "default"
+
+    @validator('url')
+    def validate_url(cls, url):
+        xyz_format = "/{z}/{x}/{y}"
+        if xyz_format not in url:
+            raise AssertionError(f"{url} needs to contain {xyz_format}")
+        return url
+
+
+class TiledImageData(BaseData):
+    """ Represents tiled imagery 
+
+    If specified version is 2, converts bounds from [lng,lat] to [lat,lng]
+
+    Requires the following args:
+        tile_layer: TileLayer
+        tile_bounds: TiledBounds
+        zoom_levels: List[int]
+    Optional args:
+        max_native_zoom: int = None
+        tile_size: Optional[int]
+        version: int = 2
+        alternative_layers: List[TileLayer]     
+
+    >>> tiled_image_data = TiledImageData(tile_layer=TileLayer,
+                                  tile_bounds=TiledBounds,
+                                  zoom_levels=[1, 12])
+    """
+    tile_layer: TileLayer
+    tile_bounds: TiledBounds
+    alternative_layers: List[TileLayer] = None
+    zoom_levels: conlist(item_type=int, min_items=2, max_items=2)
```

Comment:
I know I reccomended conlist but maybe use `Tuple[int,int]` since it is easier to read and should be immutable. Also maybe add a validator to make sure that the lower bound is less than or equal to the upper bound